modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvIterator.java

```diff
@@ -114,28 +115,58 @@ public static CsvIterator of(CharSource source, boolean headerRow, char separato
     ArgChecker.notNull(source, "source");
     @SuppressWarnings("resource")
     BufferedReader reader = Unchecked.wrap(() -> source.openBufferedStream());
+    return create(reader, headerRow, separator);
+  }
+
+  /**
+   * Parses the specified reader as a CSV file where the separator is specified and might not be a comma.
+   * <p>
+   * This factory method takes a {@link Reader}.
+   * Callers are encouraged to use {@link CharSource} instead of {@code Reader} where possible.
+   * <p>
+   * This factory method allows the separator to be controlled.
+   * For example, a tab-separated file is very similar to a CSV file, the only difference is the separator.
+   * <p>
+   * The caller is responsible for closing the reader, such as by calling {@link #close()}.
+   * 
+   * @param reader  the file reader
+   * @param headerRow  whether the source has a header row, an empty source must still contain the header
+   * @param separator  the separator used to separate each field, typically a comma, but a tab is sometimes used
+   * @return the TSV file
+   * @throws UncheckedIOException if an IO exception occurs
+   * @throws IllegalArgumentException if the file cannot be parsed
+   */
+  public static CsvIterator of(Reader reader, boolean headerRow, char separator) {
```

Comment:
It would be better to have a `Reader` version of both factory methods. Assuming most files use commas as a separator it is a bit annoying that you need to specify the separator every time if you're using a reader but not if you are using a char source. It also seems surprisingly asymmetric.